test/test_core.py

```diff
@@ -5757,14 +5757,19 @@ def test_fs_nodefs_nofollow(self):
     self.emcc_args += ['-lnodefs.js']
     self.do_runf('fs/test_nodefs_nofollow.c', 'success')
 
+  @crossplatform
   @requires_node
   def test_fs_nodefs_readdir(self):
     # externally setup an existing folder structure: existing/a
     if self.get_setting('WASMFS'):
       self.set_setting('FORCE_FILESYSTEM')
+    if not WINDOWS and not MACOS:
+      # Add an entry that isn't a directory, file, or link to test that we handle
+      # it correctly.
+      os.mkfifo(os.path.join(self.working_dir, 'named_pipe'))
```

Comment:
macos doesn't support os.mkfifo either?  TIL.

Reply:
Actually I think it is supported on MacOS I'll remove that filter.

Reply:
What if you just do `os.mkfifo('named_pipe')`?